src/Microsoft.Azure.SignalR/HubHost/ServiceLifetimeManager.cs

```diff
@@ -40,6 +42,11 @@ internal class ServiceLifetimeManager<THub> : ServiceLifetimeManagerBase<THub> w
             }
 #endif
             _clientConnectionManager = clientConnectionManager;
+
+            if (hubOptions.Value.SupportedProtocols != null && hubOptions.Value.SupportedProtocols.Any(x => x.Equals(Constants.Protocol.BlazorPack, StringComparison.OrdinalIgnoreCase)))
+            {
+                blazorDetector?.TrySetBlazor(typeof(THub).Name, true);
```

Comment:
DI `hubOptions` into `blazorDetector` and there is no need to have this `TrySetBlazor` method?

Reply:
It's hub level settings. And DI `HubOptions<THub>` will introduce `THub` and hard to use in `NegotiateHandler`.

Reply:
get it